src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs

```diff
@@ -1916,5 +1916,65 @@ private Task<string> ResetConsole()
         {
             return ConsoleCmdAsync(@"shell echo -e \\033c 1>&2");
         }
+
+        public bool MapCurrentSrcToCompilerSrc(string currentSrc, out string compilerSrc)
+        {
+            StringComparison comp = PlatformUtilities.IsWindows() ? StringComparison.OrdinalIgnoreCase : StringComparison.Ordinal;
+            foreach (var e in _launchOptions.SourceMap)
+            {
+                if (e.UseForBreakpoints && currentSrc.StartsWith(e.EditorPath, comp))
+                {
+                    var file = currentSrc.Substring(e.EditorPath.Length);
+                    if (string.IsNullOrEmpty(file)) // matched the whole directory string
+                    {
+                        break;  // use default
+                    }
+                    if (file[0] == Path.DirectorySeparatorChar || file[0] == Path.AltDirectorySeparatorChar)
+                    {
+                        file = file.Substring(1);   // Trim the directory separator
+                    }
+                    compilerSrc = Path.Combine(e.CompilerPath,file);    // map to the compiled location
```

Comment:
Will this work properly with the directory separator characters for the target OS? If you go from windows to Linux, will doing Path.Combine give you the Linux directory separators?